src/pointer-analysis/value_set.cpp

```diff
@@ -631,8 +631,8 @@ void value_sett::get_value_set_rec(
       objectt object=it->second;
 
       // adjust by offset
-      if(object.offset_is_zero() && i_is_set)
-        object.offset=i;
+      if(object.offset_is_set && i_is_set)
+        object.offset+=i;
       else
```

Comment:
Tempting, but there's a reason for what was there before.
The problem is that the above will do a concrete interpretation.
Image passing for(int i=0; i<BIG_NUM; i++) to this.

Reply:
You would need to add some flag or counter to widen when this case is hit many (=heuristic) times.

Reply:
I believe that the `insert(dest, it->first, object);` that follows this code actually takes care of this, unless I'm mistaken? That is, we seem to have taken care of this multiple times?

Reply:
Ok -- there should really be a comment there, highlighting that this is where the widening happens.